contracts/Staking.sol

```diff
@@ -79,23 +69,15 @@ contract Staking is IERC20, IERC20Metadata {
 
     function balanceOf(address _user) public view returns(uint) {
         UserInfo storage user = userInfo[_user];
-        uint acc = accamulatedRewardPerShare * user.amount - user.rewardDebt;
-        return user.accamulatedReward + acc + user.amount;
-    }
-
-    function setAdmins(address[] memory admins) public onlyOwner {
-        for(uint i = 0; i < admins.length; i++) {
-            lpAdmins.push(admins[i]);
-        }
-    }
+        uint delta = block.timestamp - lastRewardTimestamp;
+        uint updAccumulatedRewardPerShare = 
+            (delta * aprBasisPoints * calcDecimals / 10000 / 31557600) + accumulatedRewardPerShare;
```

Comment:
This expression calc (`updAccumulatedRewardPerShare`) is duplicated on line 169. Consider putting it in distinct fn.
The same with:
1) `delta` - Line 72
2) `user.accumulatedReward` - Lines 179, 193, 208

Reply:
added internal pure functions delta and calculateRewardAmount